overlord/snapshotstate/backend/backend_test.go

```diff
@@ -989,6 +990,35 @@ func (s *snapshotSuite) TestImportCheckErorr(c *check.C) {
 	c.Assert(err, check.ErrorMatches, `cannot import snapshot 14: validation failed for .+/14_foo_1.0_199.zip": snapshot entry "archive.tgz" expected hash \(d5ef563…\) does not match actual \(6655519…\)`)
 }
 
+func (s *snapshotSuite) TestImportDuplicated(c *check.C) {
+	err := os.MkdirAll(dirs.SnapshotsDir, 0755)
+	c.Assert(err, check.IsNil)
+
+	ctx := context.TODO()
+
+	epoch := snap.E("42*")
+	info := &snap.Info{SideInfo: snap.SideInfo{RealName: "hello-snap", Revision: snap.R(42), SnapID: "hello-id"}, Version: "v1.33", Epoch: epoch}
+	shID := uint64(12)
+
+	shw, err := backend.Save(ctx, shID, info, nil, []string{"snapuser"})
+	c.Assert(err, check.IsNil)
+
+	export, err := backend.NewSnapshotExport(ctx, shw.SetID)
+	c.Assert(err, check.IsNil)
+	err = export.Init()
+	c.Assert(err, check.IsNil)
+
+	buf := bytes.NewBuffer(nil)
+	c.Assert(export.StreamTo(buf), check.IsNil)
+	c.Check(buf.Len(), check.Equals, int(export.Size()))
+
+	// now import it
+	_, err = backend.Import(ctx, 123, buf)
+	dupErr, ok := err.(backend.DuplicatedSnapshotImportError)
+	c.Assert(ok, check.Equals, true)
+	c.Check(dupErr.SetID, check.Equals, shID)
```

Comment:
nitpick but my slightly preferred way to do this sort of comparison is:

```go
c.Assert(dupErr,DeepEquals,backend.DuplicatedSnapShotImportError{SetID: shID})
```

this ensures that if new fields are added that we should be checking the test will start failing if we don't update the test in conjunction too